synapse/config/oidc_config.py

```diff
@@ -243,12 +250,34 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
 # additional checks in the code.
 OIDC_CONFIG_SCHEMA = {"oneOf": [{"type": "null"}, {"type": "object"}]}
 
+# the top-level schema can contain an "oidc_config" and/or an "oidc_providers".
 MAIN_CONFIG_SCHEMA = {
     "type": "object",
-    "properties": {"oidc_config": OIDC_CONFIG_SCHEMA},
+    "properties": {
+        "oidc_config": OIDC_CONFIG_SCHEMA,
+        "oidc_providers": OIDC_PROVIDER_LIST_SCHEMA,
+    },
 }
 
 
+def _parse_oidc_provider_configs(config: JsonDict,) -> Iterable["OidcProviderConfig"]:
+    """extract and parse the OIDC provider configs from the config dict
+
+    Returns a generator which yields the OidcProviderConfig objects
```

Comment:
Is it worth mentioning that both `oidc_providers` and `oidc_config` can be provided? Or should that be an error?

Reply:
:man_shrugging: 

I'll document the current situation.